File: app/src/main/java/com/lumiyaviewer/lumiya/react/RequestQueue.java
package com.lumiyaviewer.lumiya.react;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public interface RequestQueue<K, T> extends RequestSource<K, T> {
    @Nullable
    K getNextRequest();

    @Nonnull
    ResultHandler<K, T> getResultHandler();

    void returnRequest(@Nonnull K k);

    @Nullable
    K waitForRequest() throws InterruptedException;
}

File: app/src/main/java/com/lumiyaviewer/lumiya/react/RequestQueue.kt
package com.lumiyaviewer.lumiya.react

interface RequestQueue<K, T> : RequestSource<K, T> {
    fun getNextRequest(): K?
    fun getResultHandler(): ResultHandler<K, T>
    fun returnRequest(key: K)
    @Throws(InterruptedException::class) fun waitForRequest(): K?
}
